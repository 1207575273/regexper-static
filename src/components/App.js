import React from 'react';

import Message from 'components/Message';

const App = () => (
  <Message heading="React App">
    <p>Placeholder app content</p>
  </Message>
);

export default App;
