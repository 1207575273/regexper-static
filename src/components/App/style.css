@import url('../../globals.css');

.render {
  width: 100%;
  background: var(--color-white);
  box-sizing: border-box;
  overflow: auto;
  margin: var(--spacing-margin) 0;

  & svg {
    display: block;
    transform: scaleZ(1); /* Move to separate render layer in Chrome */
    margin: 0 auto;
  }
}

.loader {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 2rem;
  background: var(--color-white);
  color: var(--color-black);

  & .message {
    font-weight: bold;
    font-size: 2.5rem;
    padding: 0;
    text-align: center;
  }

  & svg {
    display: block;
    transform: scaleZ(1); /* Move to separate render layer in Chrome */
    width: 5rem;
    height: 5rem;
    stroke: var(--color-black);
    animation: loader-spin 1s steps(8) infinite;

    & line:nth-of-type(1) { stroke: color(var(--color-black) alpha(0.75)); }
    & line:nth-of-type(3) { stroke: color(var(--color-black) alpha(0.50)); }
    & line:nth-of-type(5) { stroke: color(var(--color-black) alpha(0.25)); }
    & line:nth-of-type(7) { stroke: color(var(--color-black) alpha(0)); }
  }
}

@keyframes loader-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
