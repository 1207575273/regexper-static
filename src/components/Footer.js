import React from 'react';

const Footer = () => (
  <footer>
    <ul className="inline with-separator">
      <li>
        Created by <a href="mailto:jeff.avallone@gmail.com">Jeff Avallone</a>
      </li>
      <li>
        Generated images licensed: <a rel="license" href="http://creativecommons.org/licenses/by/3.0/">
          <img
            alt="Creative Commons CC-BY-3.0 License"
            src="https://licensebuttons.net/l/by/3.0/80x15.png" />
        </a>
      </li>
    </ul>
  </footer>
);

export default Footer;
