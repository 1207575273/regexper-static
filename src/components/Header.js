import React from 'react';
import { translate, Trans } from 'react-i18next';

import GithubIcon from 'feather-icons/dist/icons/github.svg';

const Header = () => (
  <header id="main" data-banner={ process.env.BANNER }>
    <h1>
      <a href="/">Regexper</a>
    </h1>

    <ul className="inline">
      <li><a href="https://github.com/javallone/regexper-static">
        <GithubIcon/><Trans>Source on GitHub</Trans>
      </a></li>
    </ul>
  </header>
);

export default translate()(Header);
export { Header };
