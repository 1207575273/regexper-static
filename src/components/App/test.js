jest.mock('components/SVG');

import React from 'react';
import { shallow } from 'enzyme';

import { App } from 'components/App';
import { translate } from '__mocks__/i18n';

describe('App', () => {
  test('rendering', () => {
    const component = shallow(
      <App t={ translate }/>
    );
    expect(component).toMatchSnapshot();
  });
});
