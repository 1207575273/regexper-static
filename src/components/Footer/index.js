import React from 'react';
import { translate, Trans } from 'react-i18next';

import style from './style.css';

const Footer = () => (
  <footer>
    <ul>
      <li>
        <Trans>Created by <a href="mailto:jeff.avallone@gmail.com">Jeff Avallone</a></Trans>
      </li>
      <li>
        <Trans i18nKey="Generated images licensed">
          Generated images licensed: <a rel="license external noopener noreferrer" target="_blank" href="http://creativecommons.org/licenses/by/3.0/">
            <img
              alt="Creative Commons CC-BY-3.0 License"
              src="https://licensebuttons.net/l/by/3.0/80x15.png" />
          </a>
        </Trans>
      </li>
      <li>
        <a href="/privacy.html"><Trans>Privacy Policy</Trans></a>
      </li>
    </ul>
    <div className={ style.buildId }>{ process.env.BUILD_ID }</div>
  </footer>
);

export default translate()(Footer);
export { Footer };
