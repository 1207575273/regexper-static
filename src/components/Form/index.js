import React from 'react';
import PropTypes from 'prop-types';
import { translate, Trans } from 'react-i18next';

import DownloadIcon from 'feather-icons/dist/icons/download.svg';
import LinkIcon from 'feather-icons/dist/icons/link.svg';
import ExpandIcon from 'feather-icons/dist/icons/chevrons-down.svg';

import style from './style.css';

class Form extends React.Component {
  handleSubmit = event => {
    event.preventDefault();
    this.props.onSubmit.call(this, {
      expr: this.textarea.value,
      syntax: this.syntax.value
    });
  }

  handleKeyPress = event => {
    if (event.charCode === 13 && event.shiftKey) {
      this.handleSubmit(event);
    }
  }

  textareaRef = textarea => this.textarea = textarea

  syntaxRef = syntax => this.syntax = syntax

  permalinkAction() {
    const { permalinkUrl } = this.props;

    if (!permalinkUrl) {
      return;
    }

    return <li>
      <a href={ permalinkUrl }><LinkIcon/><Trans>Permalink</Trans></a>
    </li>;
  }

  downloadActions() {
    const { downloadUrls } = this.props;

    if (!downloadUrls) {
      return;
    }

    return downloadUrls.map(({ url, filename, type, label }, i) => <li key={ i }>
      <a href={ url } download={ filename } type={ type }>
        <DownloadIcon/>{ label }
      </a>
    </li>);
  }

  render() {
    const { syntaxes, t } = this.props;

    return <div className={ style.form }>
      <form onSubmit={ this.handleSubmit }>
        <textarea
          ref={ this.textareaRef }
          onKeyPress={ this.handleKeyPress }
          autoFocus
          placeholder={ t('Enter regular expression to display') }></textarea>
        <button type="submit"><Trans>Display</Trans></button>
        <div className={ style.select }>
          <select ref={ this.syntaxRef }>
            { Object.keys(syntaxes).map(id => (
              <option value={ id } key={ id }>{ syntaxes[id] }</option>
            )) }
          </select>
          <ExpandIcon/>
        </div>
        <ul className={ ['inline', 'with-separator', style.actions].join(' ') }>
          { this.downloadActions() }
          { this.permalinkAction() }
        </ul>
      </form>

    </div>;
  }
}

Form.propTypes = {
  syntaxes: PropTypes.object,
  onSubmit: PropTypes.func,
  permalinkUrl: PropTypes.string,
  downloadUrls: PropTypes.array,
  t: PropTypes.func
};

export default translate()(Form);
export { Form };
