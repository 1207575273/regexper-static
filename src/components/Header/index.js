import React from 'react';
import { translate, Trans } from 'react-i18next';

import style from './style.css';
import GithubIcon from 'feather-icons/dist/icons/github.svg';

import LocaleSwitcher from 'components/LocaleSwitcher';

const Header = () => (
  <header className={ style.header } data-banner={ process.env.BANNER }>
    <h1>
      <a href="/">Regexper</a>
    </h1>

    <ul>
      <li><a href="https://github.com/javallone/regexper-static" rel="external noopener noreferrer" target="_blank">
        <GithubIcon/><Trans>Source on GitHub</Trans>
      </a></li>
      <li>
        <a href="/privacy.html"><Trans>Privacy Policy</Trans></a>
      </li>
      <li><LocaleSwitcher /></li>
    </ul>
  </header>
);

export default translate()(Header);
export { Header };
