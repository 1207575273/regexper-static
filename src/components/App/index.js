import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';

import style from './style.css';

import Form from 'components/Form';
import Message from 'components/Message';
import renderImage from 'components/SVG';
import { syntaxes, demoImage } from 'devel';

class App extends React.PureComponent {
  state = {
    syntaxes
  }

  setSvgUrl(element) {
    try {
      const type = 'image/svg+xml';
      const markup = element.outerHTML;
      const blob = new Blob([markup], { type });

      this.setState({
        svgUrl: {
          url: URL.createObjectURL(blob),
          label: this.props.t('Download SVG'),
          filename: 'image.svg',
          type
        }
      });
    }
    catch (e) {
      console.error(e); // eslint-disable-line no-console
    }
  }

  async setPngUrl(element) {
    try {
      const type = 'image/png';
      const markup = element.outerHTML;
      const canvas = document.createElement('canvas');
      const context = canvas.getContext('2d');
      const loader = new Image();

      loader.width = canvas.width = Number(element.getAttribute('width')) * 2;
      loader.height = canvas.height = Number(element.getAttribute('height')) * 2;

      await new Promise(resolve => {
        loader.onload = resolve;
        loader.src = 'data:image/svg+xml,' + encodeURIComponent(markup);
      });

      context.drawImage(loader, 0, 0, loader.width, loader.height);
      const blob = await new Promise(resolve => canvas.toBlob(resolve, type));

      this.setState({
        pngUrl: {
          url: URL.createObjectURL(blob),
          label: this.props.t('Download PNG'),
          filename: 'image.png',
          type
        }
      });
    }
    catch (e) {
      console.error(e); // eslint-disable-line no-console
    }
  }

  handleSubmit = ({expr, syntax}) => {
    console.log(syntax, expr); // eslint-disable-line no-console
    this.setState({
      image: demoImage
    }, async () => {
      await this.image.doReflow();
      this.setSvgUrl(this.image.svg);
      this.setPngUrl(this.image.svg);
    });
  }

  imageRef = image => this.image = image

  render() {
    const { svgUrl, pngUrl, syntaxes, image } = this.state;
    const downloadUrls = [
      svgUrl,
      pngUrl
    ].filter(Boolean);

    return <React.Fragment>
      <Form
        syntaxes={ syntaxes }
        downloadUrls={ downloadUrls }
        permalinkUrl="#permalink"
        onSubmit={ this.handleSubmit }/>
      <Message type="error" heading="Sample Error">
        <p>Sample error message</p>
      </Message>
      <Message type="warning" heading="Sample Warning">
        <p>Sample warning message</p>
      </Message>
      { image && <div className={ style.render }>
        { renderImage(image, { ref: this.imageRef }) }
      </div> }
    </React.Fragment>;
  }
}

App.propTypes = {
  t: PropTypes.func
};

export default translate()(App);
export { App };
