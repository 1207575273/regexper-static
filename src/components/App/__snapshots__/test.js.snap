// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering 1`] = `
<React.Fragment>
  <Translate(Form)
    downloadUrls={Array []}
    onSubmit={[Function]}
    permalinkUrl="#permalink"
    syntaxes={
      Object {
        "js": "JavaScript",
        "pcre": "PCRE",
      }
    }
  />
  <Message
    heading="Sample Error"
    type="error"
  >
    <p>
      Sample error message
    </p>
  </Message>
  <Message
    heading="Sample Warning"
    type="warning"
  >
    <p>
      Sample warning message
    </p>
  </Message>
</React.Fragment>
`;
